### PR TITLE
[Bug fix][Core] fixup ngram not setup correctly

### PR DESCRIPTION
**Pull request by leiwen83**
ngram_prompt_lookup_max/ngram_prompt_lookup_min need to be past through SpecDecodeWorker.create_worker's draft_worker_kwargs.

If those two doesn't get past, now there will be exception as dict cannot pop those two keys.

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @comaniac

**Review by comaniac — APPROVED** [review]
Oops. We can merge first but it should be better to add a unit test to cover this case.

**Comment by cadedaniel** [comment]
+1. Let's get a test covering this path.

**Comment by cadedaniel** [comment]
Why was it not covered by existing tests?

**Comment by comaniac** [comment]
> Why was it not covered by existing tests?

I guess existing tests directly initiated the worker, but this is more like an end-to-end path starting from a higher level?

**Comment by leiwen83** [comment]
> Why was it not covered by existing tests?

It is for current ngram still use draft model set as target model to get some info like vocab size. In this failure, ngram testcase is actually turned into multistep case with draft model same as target model...

I add a check assert in conftest to ensure we current in ngram running path, when corresponding param is set.

File: tests/spec_decode/e2e/conftest.py [hunk 1]
```diff
@@ -180,6 +179,11 @@ def get_output_from_llm_generator(
     tokens = []
     token_ids = []
     for llm in llm_generator():
+        if (not isinstance(llm, AsyncLLM)
+                and llm.llm_engine.speculative_config is not None and
+                llm.llm_engine.speculative_config.ngram_prompt_lookup_max > 0):
+            assert ("set_ngram_window_size" in dir(
+                llm.llm_engine.model_executor.driver_worker.proposer_worker))
```
Comment (comaniac) [inline tests/spec_decode/e2e/conftest.py:186]: nit: given this is just a test script.
```suggestion
            assert isinstance(llm.llm_engine.model_executor.driver_worker.proposer_worker, NGramWorker)
```

**Review by comaniac — APPROVED** [review]

**Comment by cadedaniel** [comment]
Retrying test infra failure

**Comment by comaniac** [comment]
@cadedaniel this should be able to merge.

**Review by cadedaniel — APPROVED** [review]

**Comment by simon-mo** [comment]
Spec decode tests start failing in main branch after this PR https://buildkite.com/vllm/ci/builds/6784#018f551e-d727-491c-be34-9d9fa29f4ea4

**Comment by comaniac** [comment]
The fix PR is here: #4672 
Meanwhile, @cadedaniel adjusted the test config to workaround this issue in #4592, so we should be good after merging this one.

**Pull request merged**